AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m

```diff
@@ -124,7 +133,7 @@ + (instancetype)reachabilityForInternetConnection {
 #pragma mark - Start and stop notifier
 
 - (void)startNotifier {
-  dispatch_block_t block = ^{
+  RunOnMainThread(^{
```

Comment:
Can you just add an explanation comment for why it needs to run on the main thread?

Reply:
https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1307/files/218d4e78cd9d3735c753e349c760db3f1a3e1f06#diff-961cca420883c03f21679b93b30e250fR77

The behaviour isn't changed in this PR